### PR TITLE
DT-434 Migrated to Core22/Gnome42

### PR DESCRIPTION
**Pull request by sergio-costas**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by sergio-costas** [comment]
@hellsworth This one also has incorrect data in the appdata.xml file.

File: snapcraft.yaml [hunk 1]
```diff
@@ -37,16 +41,16 @@ parts:
     source: https://gitlab.gnome.org/GNOME/gnome-mahjongg.git
     source-type: git
     source-tag: '3.38.3'
+    parse-info: [usr/share/metainfo/org.gnome.Mahjongg.appdata.xml]
     override-pull: |
-      snapcraftctl pull
-      snapcraftctl set-version $(git describe --tags --abbrev=10)
+      craftctl default
+      craftctl set version=$(git describe --tags --abbrev=10)
+      patch -p1 < $CRAFT_PROJECT_DIR/patches/patch1.diff
```
Comment (hellsworth) [inline snapcraft.yaml]: Please add a comment at the top of the patch saying what it is for. Also please give it a more intuitive name than patch1.diff. Also use *.diff here.

**Review by hellsworth — CHANGES_REQUESTED** [review]

File: snapcraft.yaml (hunk 1 above)
Comment (sergio-costas) [reply snapcraft.yaml]: Done.

**Review by hellsworth — APPROVED** [review]
Looks great thanks!

**Pull request merged**